routing/dht/routing.go

```diff
@@ -28,6 +29,8 @@ var asyncQueryBuffer = 10
 // PutValue adds value corresponding to given Key.
 // This is the top level "Store" operation of the DHT
 func (dht *IpfsDHT) PutValue(ctx context.Context, key key.Key, value []byte) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10) // CR: should this be removed instead?
```

Comment:
> Longest RTT is not more than 1s  http://www.richardclegg.org/sites/default/files/papers/ifip_nets_2012.pdf

under normal network conditions. that paper measures (typically well connected) DNS servers. Traveling around Europe, even in highly connected developed cities like Berlin, I've been routinely seeing networks with RTTs with high variances, often at 2-4s, and even up to 10s to google.com (let alone random peer X across the world). network latency varies a lot depending on the network operators attention to efficiency. And wifi access points are notoriously badly managed.